selfify/content_scripts/selfify.js

```diff
@@ -0,0 +1,49 @@
+(function() {
+  /*
+  Check and set a global guard variable.
+  If this content script is injected into the same page again,
+  it will do nothing next time.
+  */
+  if (window.hasRun) {
+    return;
+  }
+  window.hasRun = true;
+
+  /*
+  "Selfifies" the web page by:
+  * Removing every node in the document.body
+  * Inserting the selected image
+  */
+  function selfify(request, sender, sendResponse) {
+    removeEverything();
+    insertImage(request.imageURL, request.fileName);
+  }
+
+  /*
+  Remove every node under document.body
+  */
+  function removeEverything() {
+    while (document.body.firstChild) {
+      document.body.firstChild.remove();
+    }
+  }
+
+  /*
+  Given a URL to an image, create and style an iframe containing an 
+  IMG node pointing to that image, then insert the node into the document.
+  */
+  function insertImage(imageURL, fileName) {
```

Comment:
And from:
_  function selfify(request, sender, sendResponse) {
    removeEverything();
    insertImage(request.imageURL, **request.fileName**);
  }_
too